hbase-common/src/main/java/org/apache/hadoop/hbase/io/encoding/BufferedDataBlockEncoder.java

```diff
@@ -519,6 +519,7 @@ protected OffheapDecodedExtendedCell(ByteBuffer keyBuffer, short rowLength, int
     }
 
     @Override
+    @SuppressWarnings("ByteBufferBackingArray")
```

Comment:
I agree with ErrorProne that this use of a `ByteBuffer` is problematic. We're leaving it to the caller to perform bounds checking but they don't have the underlying `keyBuffer`.

Keep the warnings so that we don't lose track of this issue?